File: analysis/parser/SourceCodeParser/src/test/resources/sampleproject/bar/foo.java
package none.that.matters;

import javax.jws.WebService;

/*
 * Longer interface comment
 * that goes over multiple lines
 */
@WebService
public interface ComplexInterface extends Interface1, Interface2, Interface3 {

  // constants
  double E = 2.718282;

  // static methods
  static ZoneId getZoneId(String zoneString) {
    try {
      return ZoneId.of(zoneString);
    } catch (DateTimeException e) {
      System.err.println("Invalid time zone: " + zoneString + "; using default time zone instead.");
      return ZoneId.systemDefault();
    }
  }

  // methods
  void turn(Direction direction, double radius, double startSpeed, double endSpeed);

  int changeLanes(Direction direction, double startSpeed, double endSpeed);

  // default methods
  default boolean didItWork(int i, double x, String s) {
    return myDeck.sort(Comparator.comparing(Card::getRank)
      .reversed()
      .thenComparing(Comparator.comparing(Card::getSuit)));
  }

  // enums
  public enum Suit {
    DIAMONDS(1, "Diamonds"),
    CLUBS(2, "Clubs"),
    HEARTS(3, "Hearts"),
    SPADES(4, "Spades");

    private final int value;

    private final String text;

    Suit(int value, String text) {
      this.value = value;
      this.text = text;
    }

    public int value() {
      return value;
    }

    public String text() {
      return text;
    }
  }
}

File: analysis/analysers/parsers/UnifiedParser/src/test/resources/sampleproject/bar/foo.kt
package none.that.matters

import javax.jws.WebService
import java.time.ZoneId
import java.time.DateTimeException

/*
 * Longer interface comment
 * that goes over multiple lines
 */
@WebService
interface ComplexInterface : Interface1, Interface2, Interface3 {

  // constants
  companion object {
    const val E = 2.718282

    // static methods
    fun getZoneId(zoneString: String): ZoneId {
      return try {
        ZoneId.of(zoneString)
      } catch (e: DateTimeException) {
        System.err.println("Invalid time zone: $zoneString; using default time zone instead.")
        ZoneId.systemDefault()
      }
    }
  }

  // methods
  fun turn(direction: Direction, radius: Double, startSpeed: Double, endSpeed: Double)

  fun changeLanes(direction: Direction, startSpeed: Double, endSpeed: Double): Int

  // default methods
  fun didItWork(i: Int, x: Double, s: String): Boolean {
    return myDeck.sort(Comparator.comparing<Card, Comparable<*>> { it.rank }
      .reversed()
      .thenComparing(Comparator.comparing { it.suit }))
  }

  // enums
  enum class Suit(val value: Int, val text: String) {
    DIAMONDS(1, "Diamonds"),
    CLUBS(2, "Clubs"),
    HEARTS(3, "Hearts"),
    SPADES(4, "Spades");
  }
}
